### PR TITLE
Strip linefeeds from output

### LINKED ISSUE #1048: Ensure verify_output() handles \r correctly
``` r
library(testthat)

path <- tempfile()  
verify_output(path, cat("\r\n"))
#> Warning: Creating reference output
verify_output(path, cat("\r\n"))
#> Error: Results have changed from known value recorded in '/tmp/RtmpKf8dbf/file5e0b64a0acc5'.
#> 1/3 mismatches
#> x[2]: "\r"
#> y[2]: ""
```

<sup>Created on 2020-06-16 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

### PR DESCRIPTION
**Pull request by hadley**
Fixes #1048

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # testthat (development version)
 
+* `verify_output()` no longer always fails if output contains a line feed ("\r")
```
Comment (gaborcsardi) [inline NEWS.md:3]:
```suggestion
* `verify_output()` no longer always fails if output contains a carriage return character ("\r")
```

File: R/verify-output.R [hunk 2]
```diff
@@ -89,6 +89,7 @@ verify_output <- function(path, code, width = 80, crayon = FALSE,
     unicode = unicode,
     env = env
   )
+  output <- gsub("\r", "", output)
```
Comment (gaborcsardi) [inline R/verify-output.R:92]:
```suggestion
  output <- gsub("\r", "", output, fixed = TRUE)
```

File: tests/testthat/test-verify-output.R [hunk 3]
```diff
@@ -77,3 +77,9 @@ test_that("verify_output() doesn't use cli unicode by default", {
       cat(cli::symbol$info, cli::symbol$cross, "\n")
     })
 })
+
+test_that("verify_output() handles line-feeds", {
```
Comment (gaborcsardi) [inline tests/testthat/test-verify-output.R:81]:
```suggestion
test_that("verify_output() handles carriage return", {
```

File: tests/testthat/test-verify-output.R [hunk 4]
```diff
@@ -77,3 +77,9 @@ test_that("verify_output() doesn't use cli unicode by default", {
...
+  verify_output(test_path("test-verify-conditions-lf.txt"), {
```
Comment (gaborcsardi) [inline tests/testthat/test-verify-output.R:82]:
```suggestion
  verify_output(test_path("test-verify-conditions-cr.txt"), {
```

**Comment by gaborcsardi** [comment]
CI failure seems unrelated and a temporary network issue.

**Comment by hadley** [comment]
Oops, thanks!

**Pull request merged**